src/server/req_rescq.c

```diff
@@ -640,22 +642,40 @@ req_confirmresv(struct batch_request *preq)
 	svr_mailownerResv(presv, MAIL_CONFIRM, MAIL_NORMAL, log_buffer);
 	presv->ri_wattr[RESV_ATR_interactive].at_flags &= ~ATR_VFLAG_SET;
 
+	/* 100 extra bytes for field names, times, and count */
+	tmp_buf_size = 100 + strlen(preq->rq_user) + strlen(preq->rq_host) + strlen(next_execvnode);
+	if (tmp_buf_size > sizeof(buf)) {
+		tmp_buf = malloc(tmp_buf_size);
+		if (tmp_buf == NULL) {
+			snprintf(log_buffer, LOG_BUF_SIZE-1, "malloc failure (errno %d)", errno);
+			log_err(PBSE_SYSTEM, __func__, log_buffer);
```

Comment:
Shouldn't there be a reply to scheduler before a return? It seems that you can move code around a little bit to do what you have to do for altering reservation first and then do this buffer allocation for accounting at the end.

Reply:
you are right, moved the code.